internal/imodels/imodels.go

```diff
@@ -32,123 +32,140 @@ var osExtractors = map[string]struct{}{
-
-	Location   string // Contains Inventory.Locations[0]
-	SourceType SourceType
-
-	Commit     string
-	Repository string
+	// purlCache is used to cache the special case for SBOMs where we convert Name, Version, and Ecosystem from purls
+	// extracted from the SBOM
+	purlCache *models.PackageInfo
+	*extractor.Inventory
+}
 
-	// For package sources
-	DepGroups []string
+func (pkg *PackageInfo) Name() string {
+	// TODO(v2): SBOM special case, to be removed after PURL to ESI conversion within each extractor is complete
+	if pkg.purlCache != nil {
+		return pkg.purlCache.Name
+	}
 
-	// For OS packages
-	// This is usually the BinaryName, while Name is the SourceName
-	OSPackageName string
+	if pkg.Ecosystem().Ecosystem == osvschema.EcosystemGo && pkg.Inventory.Name == "go" {
+		return "stdlib"
+	}
 
-	AdditionalLocations []string // Contains Inventory.Locations[1..]
+	if metadata, ok := pkg.Inventory.Metadata.(*dpkg.Metadata); ok {
+		// Debian uses source name on osv.dev
+		// (fallback to using the normal name if source name is empty)
+		if metadata.SourceName != "" {
+			return metadata.SourceName
+		}
+	}
 
-	OriginalInventory *extractor.Inventory
+	return pkg.Inventory.Name
 }
 
-// FromInventory converts an extractor.Inventory into a PackageInfo.
-//
-// For ease of use, this function does not return an error, but will log
-// warnings when encountering unexpected inventory entries
-func FromInventory(inventory *extractor.Inventory) PackageInfo {
-	pkgInfo := PackageInfo{
-		Name:                inventory.Name,
-		Version:             inventory.Version,
-		Location:            inventory.Locations[0],
-		AdditionalLocations: inventory.Locations[1:],
-		OriginalInventory:   inventory,
+func (pkg *PackageInfo) Ecosystem() ecosystem.Parsed {
+	ecosystemStr := pkg.Inventory.Ecosystem()
+
+	// TODO(v2): SBOM special case, to be removed after PURL to ESI conversion within each extractor is complete
+	if pkg.purlCache != nil {
+		ecosystemStr = pkg.purlCache.Ecosystem
 	}
 
-	// Set Ecosystem
-	eco, err := ecosystem.Parse(inventory.Ecosystem())
+	// TODO: Maybe cache this parse result
+	eco, err := ecosystem.Parse(ecosystemStr)
 	if err != nil {
 		// Ignore this error for now as we can't do too much about an unknown ecosystem
 		// TODO(v2): Replace with slog
 		log.Printf("Warning: %s\n", err.Error())
 	}
-	pkgInfo.Ecosystem = eco
 
-	// Set Commit and Repository
-	if inventory.SourceCode != nil {
-		pkgInfo.Commit = inventory.SourceCode.Commit
-		pkgInfo.Repository = inventory.SourceCode.Repo
+	return eco
+}
+
+func (pkg *PackageInfo) Version() string {
+	// TODO(v2): SBOM special case, to be removed after PURL to ESI conversion within each extractor is complete
+	if pkg.purlCache != nil {
+		return pkg.purlCache.Version
 	}
 
-	// Set DepGroups
-	if dg, ok := inventory.Metadata.(scalibrosv.DepGroups); ok {
-		pkgInfo.DepGroups = dg.DepGroups()
+	return pkg.Inventory.Version
+}
+
+func (pkg *PackageInfo) SetVersion() string {
```

Comment:
This looks like it's unnecessary? It's not realy setting any values and hte contents are the same as `Verison()`.

Reply:
Ah must have been a copy and paste error, removed it now.